mod-source-record-manager-server/src/main/java/org/folio/verticle/consumers/DataImportJournalKafkaHandler.java

```diff
@@ -55,20 +55,32 @@ public DataImportJournalKafkaHandler(@Autowired Vertx vertx,
     String recordId = okapiConnectionParams.getHeaders().get(RECORD_ID_HEADER);
     Event event = new JsonObject(record.value()).mapTo(Event.class);
     LOGGER.debug("Event was received with recordId: {} event type: {}", recordId, event.getEventType());
-    String handlerBasedEventId = format("%s-%s", EVENT_ID_PREFIX, event.getId());
 
+    eventProcessedService.collectData(DATA_IMPORT_JOURNAL_KAFKA_HANDLER_UUID, event.getId(), okapiConnectionParams.getTenantId())
+      .onSuccess(res -> processJournalEvent(result, record, event, okapiConnectionParams.getTenantId()))
+      .onFailure(e -> processDeduplicationFailure(result, record, event, e));
+
+    return result.future();
+  }
+
+  private void processJournalEvent(Promise<String> result, KafkaConsumerRecord<String, String> record, Event event, String tenantId) {
     try {
-      if (!kafkaInternalCache.containsByKey(handlerBasedEventId)) {
-        kafkaInternalCache.putToCache(handlerBasedEventId);
-        DataImportEventPayload eventPayload = new ObjectMapper().readValue(event.getEventPayload(), DataImportEventPayload.class);
-        eventTypeHandlerSelector.getHandler(eventPayload).handle(journalService, eventPayload, okapiConnectionParams.getTenantId());
-      }
+      DataImportEventPayload eventPayload = new ObjectMapper().readValue(event.getEventPayload(), DataImportEventPayload.class);
+      eventTypeHandlerSelector.getHandler(eventPayload).handle(journalService, eventPayload, tenantId);
       result.complete(record.key());
     } catch (Exception e) {
       LOGGER.error("Error during processing journal event", e);
       result.fail(e);
     }
-    return result.future();
   }
 
+  private void processDeduplicationFailure(Promise<String> result, KafkaConsumerRecord<String, String> record, Event event, Throwable e) {
```

Comment:
DataImportJournalKafkaHandler does not have error handler and there is no sense to create it, because DataImportJournalKafkaHandler  should not send DI_ERROR events, so in case of duplicates exception - completes future in this handler